impl/src/bazel.rs

```diff
@@ -25,6 +25,167 @@ use crate::{
 
 use std::{env, error::Error, path::PathBuf};
 
+use cfg_expr::{targets::get_builtin_target_by_triple, Expression, Predicate};
+
+macro_rules! get_platform_triples {
+  () => {
+    [
+      // SUPPORTED_T1_PLATFORM_TRIPLES
+      get_builtin_target_by_triple("i686-apple-darwin").unwrap(),
+      get_builtin_target_by_triple("i686-pc-windows-gnu").unwrap(),
+      get_builtin_target_by_triple("i686-unknown-linux-gnu").unwrap(),
+      get_builtin_target_by_triple("x86_64-apple-darwin").unwrap(),
+      get_builtin_target_by_triple("x86_64-pc-windows-gnu").unwrap(),
+      get_builtin_target_by_triple("x86_64-unknown-linux-gnu").unwrap(),
+      // SUPPORTED_T2_PLATFORM_TRIPLES
+      get_builtin_target_by_triple("aarch64-apple-ios").unwrap(),
+      get_builtin_target_by_triple("aarch64-linux-android").unwrap(),
+      get_builtin_target_by_triple("aarch64-unknown-linux-gnu").unwrap(),
+      get_builtin_target_by_triple("arm-unknown-linux-gnueabi").unwrap(),
+      get_builtin_target_by_triple("i686-linux-android").unwrap(),
+      get_builtin_target_by_triple("i686-unknown-freebsd").unwrap(),
+      get_builtin_target_by_triple("powerpc-unknown-linux-gnu").unwrap(),
+      get_builtin_target_by_triple("s390x-unknown-linux-gnu").unwrap(),
+      get_builtin_target_by_triple("wasm32-unknown-unknown").unwrap(),
+      get_builtin_target_by_triple("x86_64-apple-ios").unwrap(),
+      get_builtin_target_by_triple("x86_64-linux-android").unwrap(),
+      get_builtin_target_by_triple("x86_64-unknown-freebsd").unwrap(),
```

Comment:
This seems like a big repetition and probably something that is worth putting in a loop with a constant list somewhere (I bet this going to be optimized away and rolled-out by the compiler).

I meant something like:
```suggestion
      [
        "i686-apple-darwin",
        "i686-pc-windows-gnu",
        "i686-unknown-linux-gnu",
        "x86_64-apple-darwin",
        "x86_64-pc-windows-gnu",
        "x86_64-unknown-linux-gnu",
        // SUPPORTED_T2_PLATFORM_TRIPLES
        "aarch64-apple-ios",
        "aarch64-linux-android",
        "aarch64-unknown-linux-gnu",
        "arm-unknown-linux-gnueabi",
        "i686-linux-android",
        "i686-unknown-freebsd",
        "powerpc-unknown-linux-gnu",
        "s390x-unknown-linux-gnu",
        "wasm32-unknown-unknown",
        "x86_64-apple-ios",
        "x86_64-linux-android",
        "x86_64-unknown-freebsd",
     ].iter().map(|x| get_builtin_target_by_triple(x).unwrap())
```

You can probably just define this list as a const and then maybe you don't even need the macro?

Reply:
I've tried to do something similar a few different ways and continue to run into 
```
calls in statics are limited to constant functions, tuple structs and tuple variants
```
I'm not sure how I can make this a global value but I do think that would be awesome. Did you have something more concrete in mind? Or maybe my declaration is just wring
```rust
static SUPPORTED_PLATFORM_TRIPLES: &'static Vec<&'static TargetInfo<'static>> = &[
    // SUPPORTED_T1_PLATFORM_TRIPLES
    "i686-apple-darwin",
     //...
    // SUPPORTED_T2_PLATFORM_TRIPLES
    "aarch64-apple-ios",
    //...
].iter()
.map(|x| {
  get_builtin_target_by_triple(x).unwrap()
})
.collect();
```

Reply:
I don't think you can do a const of a constructed type. I was alluding to having only the vector a const and maybe inline the iter().map()...

Reply:
I still don't think I understand 😅 

Your suggestion is to create one variable for the list of platforms? And to do away with the `get_platform_triples` macro? I just don't understand what the datatype of this variable is supposed to be or where it can go.

Reply:
That would look like this:

```rust
static SUPPORTED_PLATFORM_TRIPLES: &'static Vec<&'static TargetInfo<'static>> = &[
    // SUPPORTED_T1_PLATFORM_TRIPLES
    "i686-apple-darwin",
     //...
    // SUPPORTED_T2_PLATFORM_TRIPLES
    "aarch64-apple-ios",
    //...
];

...
  let support_triples = SUPPORTED_PLATFORM_TRIPLES.map(|x| {
    get_builtin_target_by_triple(x).unwrap()
  })
```

And with that you can skip the collect by chaining the action.

Reply:
Ok, I think I got it.